numpy/random/mtrand.pyx

```diff
@@ -606,9 +606,16 @@ cdef class RandomState:
             high = low
             low = 0
 
-        key = np.dtype(dtype).name
+        dt = np.dtype(dtype)
+        key = dt.name
         if key not in _integers_types:
             raise TypeError('Unsupported dtype "%s" for randint' % key)
+        if dt.byteorder != '=' and dt.byteorder != '|':
+            import warnings
+            warnings.warn('Byteorder is not supported. If you require '
+                          'platform-independent byteorder, call byteswap when '
+                          'required.\n\nIn future version, specifying '
+                          'byteorder will raise a ValueError', FutureWarning)
```

Comment:
```suggestion
                          'byteorder will raise a ValueError', DeprecationWarning)
```
Will go to error for now, so should be a DeprecationWarning.

Reply:
Maybe something more like: "Providing a dtype with a non-native byteorder is not supported." ? I think one `\n` is sufficient, but that is just nitpicking and preferences.